test/remove/remove_test.dart

```diff
@@ -169,13 +167,50 @@ void main() {
       'foo': {
         'version': '1.2.3',
         'hosted': {'name': 'foo', 'url': 'http://localhost:${server.port}'}
-      }
+      },
+      'bar': '2.0.1'
     }).create();
 
     await pubGet();
 
     await pubRemove(args: ['foo']);
-    await d.appPackagesFile({}).validate();
-    await d.appDir({}).validate();
+    await d.appPackagesFile({'bar': '2.0.1'}).validate();
+    await d.appDir({'bar': '2.0.1'}).validate();
+  });
+
+  test('preserves comments', () async {
+    await servePackages((builder) {
+      builder.serve('bar', '1.0.0');
+      builder.serve('foo', '1.0.0');
+    });
+
+    final initialPubspec = YamlDescriptor('pubspec.yaml', '''
+      name: myapp
+      dependencies: # comment A
+          # comment B
+          bar: 1.0.0 
+          foo: 1.0.0 # comment C
+        # comment D
+    ''');
+    await d.dir(appPath, [initialPubspec]).create();
+
+    await pubGet();
+
+    await pubRemove(args: ['bar']);
+
+    final finalPubspec = YamlDescriptor('pubspec.yaml', '''
+      name: myapp
+      dependencies: # comment A
+          # comment B
+          foo: 1.0.0 # comment C
+        # comment D
+    ''');
+    await d.dir(appPath, [finalPubspec]).validate();
+    final fullPath = p.join(d.sandbox, appPath, 'pubspec.yaml');
+
+    expect(File(fullPath).existsSync(), true);
+
+    final contents = File(fullPath).readAsStringSync();
+    expect(contents, await finalPubspec.read());
```

Comment:
nit: can't we make this a bit more robust by simply doing:
`expect(contents, contains('# comment A'));` and similar for B, C and D,

That way we're not relying minor whitespace changes in `package:yaml_edit`, but we are checking that comments are preserved.

Reply:
hm okay. I initially put it this way because I thought something like:
```
      name: myapp
      dependencies: # comment # comment B
          foo: 1.0.0 # comment C
        # comment D
```

could realistically happen but is not what we want, but you have a point. fixing now